### PR TITLE
feat: recovery info modals + widgets

### LINKED ISSUE #2758: [Recovery] Proposal request modal/widget
## What is the feature about

When a recoverer connects to a Safe, a full-screen modal is displayed requesting a recovery attempt. If "I'll do it later" is clicked, the decision is cached and a dashboard widget shown in its place.

## The list of requirements

- [ ] Create full-screen modal with caching on decline
- [ ] Create dashboard widget

## Designs/sketches

image

image

## Links

- [Feature spec.](https://www.notion.so/safe-global/Recovery-feature-spec-d873c33a27c24cfabc91c85dbcf3ea35)
- [Figma for modal](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Generic-recovery?type=design&node-id=505-26586&mode=design&t=Lo3nDvw622nW8RA7-0)
- [Figma for widget](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Generic-recovery?type=design&node-id=540-11779&mode=design&t=Lo3nDvw622nW8RA7-0)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #2758 and #2759

## How this PR fixes it

This adds full-screen modals that appear if/when a recovery proposal is either added to the queue (for an owner/guardian) or when none is queued (for guardian).

## How to test it

1. Open a Safe with recovery enabled, ensuring no recovery transactions are queued. As a guardian, observe the "proposal" modal.
2. Open a Safe with recovery enabled, ensuring no recovery transactions are queued. As a owner/guardian, observe the "in-progress" modal.

## Screenshots

image

image

image

image

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/dashboard/RecoveryHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+// Appease TypeScript
+const _useSupportedRecovery = () => useHasFeature(FEATURES.RECOVERY)
+
+export const RecoveryHeader = madProps(_RecoveryHeader, {
+  isOwner: useIsSafeOwner,
+  isGuardian: useIsGuardian,
+  supportsRecovery: _useSupportedRecovery,
+  queue: useRecoveryQueue,
+})
```
Comment (katspaugh) [inline src/components/dashboard/RecoveryHeader/index.tsx:57]: 👍

File: src/components/recovery/RecoveryCards/RecoveryProposalCard.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        <Grid item alignSelf="flex-end">
+          <Button onClick={onClose}>I&apos;ll do it later</Button>
+          {recoveryButton}
+        </Grid>
+      </Grid>
+    </Card>
+  )
+}
+
+// Appease TypeScript
```
Comment (katspaugh) [inline src/components/recovery/RecoveryCards/RecoveryProposalCard.tsx:111]: Is it because it wants to have a name?

**Review by katspaugh — APPROVED** [review]
Please make sure the icons work in dark mode.

File: src/components/recovery/RecoveryCards/RecoveryProposalCard.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/recovery/RecoveryCards/RecoveryProposalCard.tsx:111]: It complains about the rules of hooks as `madProps` is not a hook or component. I'd sooner extract it to a "middleman" hook than disabling the rule.

image

**Comment by iamacook** [comment]
> Please make sure the icons work in dark mode.

I've made @TanyaEfremova aware of this. I'm going to merge this in the mean time and we can revisit it.

**Pull request merged**